src/services/refactors/generateGetAccessorAndSetAccessor.ts

```diff
@@ -0,0 +1,165 @@
+/* @internal */
+namespace ts.refactor.generateGetAccessorAndSetAccessor {
+    const actionName = "Generate 'get' and 'set' accessors";
+    const actionDescription = Diagnostics.Generate_get_and_set_accessors.message;
+    registerRefactor(actionName, { getEditsForAction, getAvailableActions });
+
+    interface Info {
+        originalName: string;
+        fieldName: string;
+        accessorName: string;
+        accessorType: TypeNode;
+        propertyDeclaration: PropertyDeclaration;
+        needUpdateName: boolean;
+        hasModifiers: boolean;
+        needUpdateModifiers: boolean;
+    }
+
+    function getAvailableActions(context: RefactorContext): ApplicableRefactorInfo[] | undefined {
+        const { file, startPosition } = context;
+
+        const fieldInfo = getConvertibleFieldAtPosition(file, startPosition);
+        if (!fieldInfo) return undefined;
+
+        return [
+            {
+                name: actionName,
+                description: actionDescription,
+                actions: [
+                    {
+                        name: actionName,
+                        description: actionDescription
+                    }
+                ]
+            }
+        ];
+    }
+
+    function getEditsForAction(context: RefactorContext, _actionName: string): RefactorEditInfo | undefined {
+        const { file, startPosition } = context;
+
+        const fieldInfo = getConvertibleFieldAtPosition(file, startPosition);
+        if (!fieldInfo) return undefined;
+
+        const changeTracker = textChanges.ChangeTracker.fromContext(context);
+        const newLineCharacter = getNewLineOrDefaultFromHost(context.host, context.formatContext.options);
+
+        const { fieldName, accessorName, accessorType, propertyDeclaration, needUpdateName, hasModifiers, needUpdateModifiers } = fieldInfo;
+        const accessorModifiers = hasModifiers ? (
+            (getModifierFlags(propertyDeclaration) & ModifierFlags.Private || !propertyDeclaration.modifiers) ? createNodeArray([createToken(SyntaxKind.PublicKeyword)]) : propertyDeclaration.modifiers
+        ) : undefined;
+
+        const getAccessor = generateGetAccessor(fieldName, accessorName, accessorType, accessorModifiers);
+        const setAccessor = generateSetAccessor(fieldName, accessorName, accessorType, accessorModifiers);
+
+        const modifiers = hasModifiers ? createNodeArray([createToken(SyntaxKind.PrivateKeyword)]) : undefined;
+        if (needUpdateName || needUpdateModifiers) {
+            changeTracker.replaceNode(file, propertyDeclaration, updateoriginalPropertyDeclaration(propertyDeclaration, fieldName, modifiers), {
+                suffix: newLineCharacter

```

Comment:
With the latest changes to `ChangeTracker` it shouldn't be necessary to explicitly specify options. If it is, file an issue and we can fix it later after this PR is merged.